src/html/mod.rs

```diff
@@ -9,3 +12,63 @@ pub use self::local_name::{LocalName, LocalNameHash};
 pub use self::namespace::Namespace;
 pub use self::tag::Tag;
 pub use self::text_type::TextType;
+
+/// Convert text to HTML
+#[inline]
+pub(crate) fn escape_body_text(mut content: &str, output_handler: &mut impl FnMut(&str)) {
+    loop {
+        if let Some(pos) = memchr3(b'&', b'<', b'>', content.as_bytes()) {
+            let Some((chunk_before, (matched, rest))) = content
+                .split_at_checked(pos)
+                .and_then(|(before, rest)| Some((before, rest.split_at_checked(1)?)))
```

Comment:
Fluent syntax is quite hard to read here. Let's use classic control flow here instead